src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        int index = random.nextInt(5);
+        ColorSupplier colorSupplier = new ColorSupplier();
+
+        Circle circle = new Circle();
+        IsoscelesTrapezoid trapezoid = new IsoscelesTrapezoid();
+        Rectangle rectangle = new Rectangle();
+        RightTriangle rightTriangle = new RightTriangle();
+        Square square = new Square();
+
+        switch (index) {
+            case 0: {
+                circle.setRadius(random.nextDouble(100.0));
+                circle.setColor(colorSupplier.getRandomColor());
+                return circle;
+            }
+            case 1: {
+                trapezoid.setHeight(random.nextDouble(10.0,20.0));
+                trapezoid.setFirstBase(random.nextDouble(10.0,20.0));
+                trapezoid.setSecondBase(random.nextDouble(15.0, 30.0));
+                trapezoid.setColor(colorSupplier.getRandomColor());
+                return trapezoid;
```

Comment:
No reason to write return in all cases, it is possible to define method level variable and populate it and return it the end of method